tools/filesystem.lua

```diff
@@ -2,52 +2,58 @@ local module = {}
 
 module.WINDOWS = 0
 module.LINUX = 1
+module.MACOS = 2
 local platform = -1
 
-local function detectPlatform()
-  if platform > -1 then
-    return
+local function run(command, failure_msg)
+  local handle = io.popen(command)
+  local output = handle:read("*a") -- returns "" if no stdout
+  local success, _, ret_code = handle:close()
+  if failure_msg and ret_code ~= 0 then
+    assert(false, "RUN: '"..command.."' failed. "..failure_msg)
   end
+  return output, ret_code
+end
 
-  local command = io.popen("ver")
-  local output = command:read("*a")
-  if output and string.match(output, "Microsoft Windows") then
-    platform = module.WINDOWS
-  else
-    -- TODO: actually do a test for Linux - does linux have a ver command?
-    platform = module.LINUX
+local function detectPlatform()
+  if platform == -1 then
+    if package.config:sub(1,1) == '\\' then
+      platform = module.WINDOWS
+    else
+      local uname, _ = run("uname -s")
```

Comment:
Since we're not using the return code, let's not create a var for it:
```suggestion
      local uname = run("uname -s")
```